code/datums/components/blocker.dm

```diff
@@ -0,0 +1,162 @@
+		2. Block Poor: Provide a minimum credit value for the argument, if someone is wearing an ID with an account that has less than that amount, or failing that has their face unobscured and has a registered account with less
+			than that value, they will be blocked. Not sure if I should block those without a linked account, probably won't. Maybe let it scan their inventory to see if they have cash in their bag or pockets?
+		3. Block All: Everything gets blocked! Mostly debug for now
+
+*/
+
+#define BLOCK_JOBS	0
+#define BLOCK_POOR 	1
+#define BLOCK_ALL 	2
+
+#define REFRESH_EVERY	5 SECONDS
+
+/datum/component/blocker
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+	var/list/block_list = list()
+	var/list/image/staticOverlays = list()
+
+	var/min_cash = 100
+	var/list/blocked_jobs = list("Assistant")
+	var/mode = BLOCK_JOBS
+	var/last_update = 0
+
+
+/datum/component/blocker/Initialize(block_mode, arg)
+	if(!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	mode = block_mode
+
+	switch(mode)
+		if(BLOCK_JOBS)
+			if(!islist(arg))
+				return COMPONENT_INCOMPATIBLE
+			blocked_jobs = arg
+		if(BLOCK_POOR)
+			min_cash = arg
+
+	START_PROCESSING(SSdcs, src)
+
+/datum/component/blocker/Destroy(force, silent)
+	return ..()
+
+/datum/component/blocker/RegisterWithParent()
+
+/datum/component/blocker/UnregisterFromParent()
+
+/// This proc registers signals to override examining blocked entities so we won't be able to tell what they are, and is undone by clear_blocklist()
+/datum/component/blocker/proc/block_examines()
+	for(var/mob/living/L in block_list)
+		RegisterSignal(L, COMSIG_PARENT_EXAMINE, .proc/block_examine)
+		RegisterSignal(L, COMSIG_ATOM_GET_EXAMINE_NAME, .proc/block_examine_name)
+
+/// This proc will block examine text to hide what the target actually is
+/datum/component/blocker/proc/block_examine(datum/source, mob/user, list/examine_list)
+	if(user == parent)
+		examine_list = list("</span class='notice'>You can't make out anything behind all the static!</span>")
+		return examine_list
+
+/datum/component/blocker/proc/block_examine_name(datum/source, mob/user, list/override)
+	if(user == parent)
+		override[EXAMINE_POSITION_BEFORE] = " \[BLOCKED\] " // todo: actually have it overwrite the name
+		return COMPONENT_EXNAME_CHANGED
+
+/// Clear out the static overlays, unhook the examine block signals, and
+/datum/component/blocker/proc/clear_blocklist()
+	var/mob/living/P = parent
+	if(P && P.client)
+		for(var/image/static in staticOverlays)
+			P.client.images.Remove(static)
+
+	staticOverlays = list()
+
+	for(var/mob/living/L in block_list)
+		UnregisterSignal(L, COMSIG_PARENT_EXAMINE)
+		UnregisterSignal(L, COMSIG_ATOM_GET_EXAMINE_NAME)
+
+	block_list = list()
+
+/// Based on the mode and args we have set, generate block_list so we know who we don't want to see
+/datum/component/blocker/proc/generate_blocklist()
```

Comment:
This gets run every few seconds for every single person that has this quirk. Quite frankly, this is ridiculous. And I thought I was shitcoding by making digital camo process.
You need to either come up with a really cheap way of implementing this instead or this just isn't viable.

Reply:
Yeah this is just napkin code, I'm planning to look at how sec/medhuds handle checking tonight. At the very least I'll make it so it only has to remove/make icons for new changes rather than all of em